## Transcribing DNA into RNA
Link: https://rosalind.info/problems/rna/

In [9]:
def transcribe_dna_2_rna(dna_string):
    if not dna_string:
        return "Invalid String"
    dna_string = dna_string.upper()
    return dna_string.replace('T', 'U')

In [10]:
from utils import dna_utils

In [11]:
dna_string = dna_utils.read_dna_from_txt_file("../data/rosalind_rna.txt")

In [12]:
print(dna_utils.benchmark(transcribe_dna_2_rna, dna_string))

function transcribe_dna_2_rna: time: 0.009059906005859375, dna_length: 924


In [13]:
print(dna_utils.benchmark(transcribe_dna_2_rna, dna_utils.create_dna_string(10000)))

function transcribe_dna_2_rna: time: 0.06198883056640625, dna_length: 10000


In [14]:
print(dna_utils.benchmark(transcribe_dna_2_rna, dna_utils.create_dna_string(100000)))

function transcribe_dna_2_rna: time: 0.6041526794433594, dna_length: 100000


In [15]:
print(dna_utils.benchmark(transcribe_dna_2_rna, dna_utils.create_dna_string(1000000)))

function transcribe_dna_2_rna: time: 5.118131637573242, dna_length: 1000000


In [16]:
print(dna_utils.benchmark(transcribe_dna_2_rna, dna_utils.create_dna_string(10000000)))

function transcribe_dna_2_rna: time: 46.15283012390137, dna_length: 10000000


# Benchmark Results:
    function transcribe_dna_2_rna: time: 0.007867813110351562, dna_length: 924
    function transcribe_dna_2_rna: time: 0.04601478576660156, dna_length: 10000
    function transcribe_dna_2_rna: time: 0.5469322204589844, dna_length: 100000
    function transcribe_dna_2_rna: time: 4.536867141723633, dna_length: 1000000
    function transcribe_dna_2_rna: time: 53.64203453063965, dna_length: 10000000

## Reasoning: Python string methods are implemented in C, and hence are much faster than a Python implementation from scratch.